python/tank/authentication/unified_login_flow2.py

```diff
@@ -41,11 +46,15 @@ def process(
     browser_open_callback=None,
     keep_waiting_callback=lambda: True,
 ):
-    sg_url = connection.sanitize_url(sg_url)
-
-    if not product and "TK_AUTH_PRODUCT" in os.environ:
-        product = os.environ["TK_AUTH_PRODUCT"]
+    """
+    :param product: **mandatory** - string
```

Comment:
I'm not sure if this is going to be used on the generated docs but won't it cause any issue if we don't document the remaining arguments?

Also, I see in this line `product` is mandatory. However, it can be defaulted to `None` according to the function signature.